lua/kommentary/init.lua

```diff
@@ -5,49 +5,98 @@ This module handles the initialization of the plugin.
 ]]
 local kommentary = require("kommentary.kommentary")
 local config = require("kommentary.config")
+local util = require("kommentary.util")
 local M = {}
+local context = util.enum({"line", "visual", "motion", "init"})
+local modes = config.get_modes()
 
 --[[--
-Toggle comment on current line, selection or motion.
+Function to be called by mappings.
+This acts as a middle-man, when called it will figure out the context of the
+call, so if it was called on a single line, or from a selection/motion, and
+also the range of the operation, so on which line to start and end.
+It will then call another function which takes care of calling the functions
+in kommentary.kommentary, which then actually comments out the range.
+You can overload this function with another function as an argument, which
+will overwrite the function that gets called at the end. The arguments
+passed to the second function are: line_number_start, line_number_end, context
```

Comment:
```suggestion
Function to be called by mappings.
This function should be called by all the mappings, it will figure out two things:
    * The context of the call, meaning if the mapping was triggered in on a
      single line, or from a selection/motion
    * The range of the operation, meaning on which line the operation starts and ends.
This function does not, however, manipulate the buffer in any way, that is left up to
other functions to allow for greater customizability. At the end of this function, it
will call a *callback* function with the following arguments:
    line_number_start, line_number_end, context
This callback function, by default, will be `kommentary.toggle_comment`, and will
toggle the range specified. If you wish to use a different function, you can provide
it in a table as the third argument to this function `kommentary.go`, for example, if
the desired callback function were called `kommentary.toggle_comment_singles`,
then you would call this function like this: 
    `kommentary.go(context, {kommentary.toggle_comment_singles})`
```

This explanation is a little confusing.